src/Stratis.Bitcoin.Features.PoA/Voting/VotingManager.cs

```diff
@@ -548,26 +579,11 @@ private void ProcessBlock(DBreeze.Transactions.Transaction transaction, ChainedH
 
                             if (poll == null)
                             {
-                                // Ensures that highestPollId can't be changed before the poll is committed.
-                                this.PollsRepository.Synchronous(() =>
-                                {
-                                    poll = new Poll()
-                                    {
-                                        Id = this.PollsRepository.GetHighestPollId() + 1,
-                                        PollVotedInFavorBlockData = null,
-                                        PollExecutedBlockData = null,
-                                        PollStartBlockData = new HashHeightPair(chBlock.ChainedHeader),
-                                        VotingData = data,
-                                        PubKeysHexVotedInFavor = new List<Vote>() { new Vote() { PubKey = fedMemberKeyHex, Height = chBlock.ChainedHeader.Height } }
-                                    };
-
-                                    this.polls.Add(poll);
+                                if (data.Key == VoteKey.AddFederationMember)
```

Comment:
Just to understand this again, why are we only doing this with add member polls?

Reply:
The pending poll is now being created in response to a valid the voting request. That allows us to add the restriction that we only process "add member" votes if the pending poll already exists.